ibm/data_source_ibm_cr_namespaces.go

```diff
@@ -54,38 +81,42 @@ func dataIBMContainerRegistryNamespaces() *schema.Resource {
 	}
 }
 
-func dataIBMContainerRegistryNamespacesRead(d *schema.ResourceData, meta interface{}) error {
-	userDetails, err := meta.(ClientSession).BluemixUserDetails()
-	if err != nil {
-		return err
-	}
-	accountID := userDetails.userAccount
-
-	crClient, err := meta.(ClientSession).ContainerRegistryAPI()
+func dataIBMContainerRegistryNamespacesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	containerRegistryClient, err := meta.(ClientSession).ContainerRegistryV1()
 	if err != nil {
-		return err
-	}
-	target := registryv1.NamespaceTargetHeader{
-		AccountID: accountID,
+		return diag.FromErr(err)
 	}
 
-	crAPI := crClient.Namespaces()
+	listNamespaceDetailsOptions := &containerregistryv1.ListNamespaceDetailsOptions{}
 
-	response, err := crAPI.GetDetailedNamespaces(target)
+	namespaceDetailsList, response, err := containerRegistryClient.ListNamespaceDetails(listNamespaceDetailsOptions)
 	if err != nil {
-		return err
+		if response != nil && response.StatusCode == 404 {
```

Comment:
in datasource don't addd code to check if response != nil && response.StatusCode == 404 { ----}
just return back the err

Reply:
Hmm - I think this was one of Mike's hand-edits. This is the case where the account doesn't have any namespaces in that region.. @mkistler - I'm trying to recall why this was needed?

Reply:
I checked with Mike - and he agrees with you. I made the change and have tested it.